iaso/models/org_unit.py

```diff
@@ -15,6 +15,26 @@
 from .project import Project
 
 
+def get_or_create_org_unit_type(name: str, project: Project, depth: int) -> "OrgUnitType":
+    # Logic is very messy, but attempts to simplify were unsuccessful because there's messy data in production
+    # (maybe synced with other devices).
+    # This function has probably the potential to create more messy data, so I still suggest to refactor CAREFULLY.
+    out = OrgUnitType.objects.filter(projects=project, name=name).first()
+    if not out:
+        count = OrgUnitType.objects.filter(name=name, short_name=name[:4], depth=depth).count()
+        if count == 0:
+            out = OrgUnitType.objects.create(name=name, short_name=name[:4], depth=depth)
+            out.save()
+        elif count > 1:
+            out = OrgUnitType.objects.filter(name=name, short_name=name[:4], depth=depth, projects=project).first()
```

Comment:
I think that's where us the original bug the `projects=project`, if condition of line 22 is not true there is no way this one is true

Reply:
Just removing the `elif count > 1`, wont work, because `get()` will be called in the else branch when there are multiple matching OUT. We can also adjust that else branch, but then we're back at simplifying and adjusting the logic. That's what I did initially, but I had to revert it because of the messy data in production... It feels a bit like we're stuck in a loop here :/